### PR TITLE
Enhancement/programming online editor UI

### PR DESCRIPTION
**Pull request by Malyuk-A**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I added multiple integration tests related to the features (with a high test coverage)

### Motivation and Context
This is another follow-up PR concerning the online editor user interface for programming exercises.
(These are the previous ones:  https://github.com/ls1intum/Artemis/pull/3843 & https://github.com/ls1intum/Artemis/pull/3832)

This PR implements 2 changes :

1) Hides the draggable icons "||" between the left/right and middle window. These have been shown if the window (e.g. problem statement) was minimized. You could and should not be able to change the size of the minimized window.

2) Changes the look of the minimized file browser to be similar to the problem-statement/instruction window.
<image src="https://user-images.githubusercontent.com/65814168/127891864-34f4ba4e-7aab-4314-9c16-59bc7a1844a9.png" 
alt="before" width="250"/>
<image src="https://user-images.githubusercontent.com/65814168/127892721-478f2239-7b05-4183-95f8-18b90a349057.png" 
alt="after" width="250"/>

_3) additionally added the respective icon to the minimized view (requested by Stephan)_

### Description

Analog changes/implementation to the previous PRs.

### Steps for Testing

Find or create a programming exercise.
Open the online code editor.
Check if the file browser's new look works and is similar to the problem-statement/instruction window.

P.S. @SimonEntholzer : Thank you very much for helping me to install the local programming setup. 🐧👍

### CODE REVIEW

**Comment by Malyuk-A** [comment]
I decided to open this PR for review because the client tests started working after I merged develop into this brach.
The reason for the failing of client-tests-fast seems to be an external issue ( Dominik experiences the same problems with a totally different PR)

**Review by lschlesinger — COMMENTED** [review]
It looks very good now! Tested on Ts2, problem statement and file browser look the same and behave the same.
I just recognized, that the draggable icons || appear below the build output is it is collapsed (see picture) - so this should be changed in order to be consistent with the problem statement and the file browser which do not show these icons when they are collapsed. Could you apply the same changes on this draggable/collapsable build output as well?
Bildschirmfoto 2021-08-07 um 11 21 04

**Review by lschlesinger — COMMENTED** [review]
Mentioned UI "inconsistency" for Build Output is fixed. I have another comment, that could be addressed easily. The icons marked in the picture do not have the class `clickable` to give the cursor the CSS value `pointer` and indicate, that it is clickable. You can try it yourself to see, what I mean. The other arrows change the cursor, so it's another minor inconsistency.
Bildschirmfoto 2021-08-08 um 18 11 02

**Review by FrankeLukas — APPROVED** [review]
Works as advertised.

**Comment by Malyuk-A** [comment]
Well... I have merged develop a couple of times into this branch and the last couple of times Client Tests passed without any error. Today I merged develop again and now 1 test is failing for me in `course-overview.component`
image

The last time this file has been modified (if I am not mistaken) was 2 months ago.
I did not touch the failing course-overview component and this test was successful before, so ... should I try to fix it, or is this some kind of strange mistake?

**Review by lschlesinger — DISMISSED** [review]
Works as described, tests on TS2

**Review by FrankeLukas — APPROVED** [review]
Re-tested, so re-approve.

**Review by SimonEntholzer — CHANGES_REQUESTED** [review]
Code looks really good, only some very minor things. :)

File: src/main/webapp/app/exercises/programming/shared/code-editor/instructions/code-editor-instructions.component.html [hunk 1]
```diff
@@ -11,13 +11,14 @@ <h3 class="card-title">
...
+                <fa-icon class="top-icon" [icon]="'chevron-left'"></fa-icon>
+                <fa-icon [icon]="['far', 'list-alt']" class="fa-rotate-90" id="panel-symbol"></fa-icon>
                 <span *ngIf="!isAssessmentMode" class="header" jhiTranslate="artemisApp.exercise.problemStatement">Problem Statement</span>
                 <span *ngIf="isAssessmentMode" class="header" jhiTranslate="artemisApp.exercise.instructions">Instructions</span>
-                <fa-icon [icon]="'chevron-left'"></fa-icon>
+                <fa-icon class="bottom-icon" [icon]="'chevron-left'"></fa-icon>
             </span>
         </h3>
-        <fa-icon *ngIf="!collapsed" class="ms-auto" [icon]="'chevron-right'"></fa-icon>
+        <fa-icon *ngIf="!collapsed" class="ms-auto clickable" [icon]="'chevron-right'"></fa-icon>
```
[the remark this answers is not in the document]
Comment (julian-christl) [reply src/main/webapp/app/exercises/programming/shared/code-editor/instructions/code-editor-instructions.component.html:21]: Just a guess tbh but I assume this was the request Lorena had. Something along the lines of
```css
.clickable {
    cursor: pointer;
}
```

[the remark this answers is not in the document]
Comment (Malyuk-A) [reply src/main/webapp/app/exercises/programming/shared/code-editor/instructions/code-editor-instructions.component.html:21]: Good question. As far as I understood this there was a difference if you hovered with your mouse courses above the "<" closing/collapse icon when the respective window was collapsed (here your mouse pointer turned into 👆 (a finger, i.e. indicator that this button is clickable). But when the window was opened (not collapsed) there was no such behavior (change to 👆). I changed this by adding the class "clickable" now also when the window is open the cursor changes to 👆.

File: src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.component.html [hunk 2]
```diff
@@ -20,18 +23,23 @@
 
             <!-- Instructions -->
             <div class="editor-sidebar-right collapsable">
-                <fa-icon [icon]="'grip-lines-vertical'" class="rg-sidebar-right text-lightgrey"><span></span></fa-icon>
+                <fa-icon *ngIf="!rightPanelIsCollapsed" [icon]="'grip-lines-vertical'" id="draggableIconForInstructions" class="rg-sidebar-right text-lightgrey"
+                    ><span></span
+                ></fa-icon>
```
[the remark this answers is not in the document]
Comment (Malyuk-A) [reply src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.component.html:28]: Yes, I also prefer it like you have said. This is a strange result of using prettier:write

[the remark this answers is not in the document]
Comment (Malyuk-A) [reply src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.component.html:28]: Yes, I also prefer it like you have said. This is a strange result of using prettier:write

File: src/test/javascript/spec/component/code-editor/code-editor-grid.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    beforeEach(() => {
+        return TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [CodeEditorGridComponent],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CodeEditorGridComponent);
+                comp = fixture.componentInstance;
+                debugElement = fixture.debugElement;
```
[the remark this answers is not in the document]
Comment (Malyuk-A) [reply src/test/javascript/spec/component/code-editor/code-editor-grid.component.spec.ts:33]: Thank you for noticing

File: src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.component.html [hunk 4]
```diff
@@ -10,7 +10,10 @@
             <div class="editor-sidebar-left collapsable">
                 <ng-content select="[editorSidebarLeft]"></ng-content>
                 <!-- Required for resizing; don't remove empty span -->
-                <fa-icon [icon]="'grip-lines-vertical'" class="rg-sidebar-left text-lightgrey"><span></span></fa-icon>
+                <fa-icon *ngIf="!fileBrowserIsCollapsed" [icon]="'grip-lines-vertical'" id="draggableIconForFileBrowser" class="rg-sidebar-left text-lightgrey">
+                    <span></span>
```
Comment (MerlinMehmed) [inline src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.component.html:14]: only for my understanding - why is the empty span needed here? It also appears in the other icons in this component.

**Review by MerlinMehmed — APPROVED** [review]
Tested on TS1, works as expected. The code also looks good

File: src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.component.html (hunk 4 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.component.html:14]: I think this is needed for the correct placement/fontsize of the icon for some reason, but I honestly can't tell you why. I guess we could check if it still works without it

**Review by SimonEntholzer — APPROVED** [review]
Code looks good now, I did not test it though.

**Review by jpbernius — APPROVED** [review]

**Pull request merged**